### ISSUE TITLE
Synchronize with Github only when a new release is created

### ISSUE THREAD

**Issue by devnoname120** [JasonBarnabe/greasyfork#361]
Would it be possible to add an option, so that Greasyfork only pulls changes when a new release is issued?

**Comment by JasonBarnabe** [comment]
Well I guess the question would be which releases would trigger the webhook? All of them? I could see people potentially having multiple branches of releases and not wanting 1.1.1 to overwrite what happened with 1.2.0.

**Comment by devnoname120** [comment]
On Github the latest release is the release containing the most recent commit.
It is uncommon to create a release for another branch than master.
But I think that we should stick to the current GreasyFork behavior for synchronization, namely only check what happens on the master branch.

Tl;Dr: Only care about releases on master, only take the one with the most recent commit into consideration.

**Comment by devnoname120** [comment]
Maybe add a field for the name of the branch? (Default: master.)

**Comment by JasonBarnabe** [comment]
I was referring to branches of releases, like 1.1.1 vs 1.2.0. Branches in GitHub are a different concept.

**Comment by aaaxx** [comment]
Any news here? Can't you just pull if the version number is greater than the current one?

**Comment by devnoname120** [comment]
@aaaxx I suggest you to have a branch `develop` that you only merge to master when doing a release. This is what I did.

**Comment by JasonBarnabe** [comment]
6 years later, this is now supported for GitHub and GitLab. Instructions can be found [here](https://greasyfork.org/en/users/webhook-info).

**Comment by eight04** [comment]
How does greasyfork handle code update? For example:

1. Import a script and setup the web hook
2. Change the code in the repository without updating the version number
3. Sync from web hook  

The result will be

A. The code is updated so there are two different code with same version number in the history
B. Nothing happens
C. Others?

**Comment by JasonBarnabe** [comment]
If the version number is not incremented when a script is updated from a webhook, the new code will be saved anyway.

**Comment by eight04** [comment]
Tried pushing a tag to Github and it seems that the tag doesn't trigger the hook. Do we have to create the release manually i.e. https://github.com/eight04/ptt-imgur-fix/releases/new?

**Comment by JasonBarnabe** [comment]
This happens on release, not just tag.
